usr/lib/cca_stdll/cca_specific.c

```diff
@@ -8656,10 +9865,53 @@ static CK_RV cca_handle_apqn_event(STDLL_TokData_t *tokdata,
         __sync_or_and_fetch(&cca_private->inconsistent, TRUE);
         TRACE_ERROR("CCA master key setup is inconsistent, all crypto operations will fail from now on\n");
         OCK_SYSLOG(LOG_ERR, "CCA master key setup is inconsistent, all crypto operations will fail from now on\n");
-    } else {
-        __sync_and_and_fetch(&cca_private->inconsistent, FALSE);
+        return CKR_OK;
     }
 
+    __sync_and_and_fetch(&cca_private->inconsistent, FALSE);
+
+    /* Re-check after APQN set change if protected key support is available */
+    rc = cca_get_min_card_level(tokdata);
+    if (rc != CKR_OK) {
+        TRACE_WARNING("Could not re-determine min card level, protected key support not available.\n");
+        return rc;
+    }
+
+#ifndef NO_PKEY
+    {
```

Comment:
Please avoid this block and move the min_card_version variable declaration (incl #ifdef around) to the top of the function.

Reply:
Yes, this was my initial approach, but when compiled with NO_PKEY we get an "unused variable" warning. So another #ifndef is necessary around the variable .. but will fix.

Reply:
Done.